src/test/java/ibm/jceplus/junit/base/BaseTestResetByteBuffer.java

```diff
@@ -0,0 +1,136 @@
+            Arguments.of("AES/GCM/NoPadding", false, true, true, false),
+            Arguments.of("AES/GCM/NoPadding", false, true, true, false),
+            Arguments.of("AES/GCM/NoPadding", false, true, false, true),
+            Arguments.of("AES/GCM/NoPadding", false, true, false, true),
+            Arguments.of("AES/GCM/NoPadding", false, true, false, false),
+            Arguments.of("AES/GCM/NoPadding", false, true, false, false),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, true, true),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, true, true),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, true, false),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, true, false),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, false, true),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, false, true),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, false, false),
+            Arguments.of("AES/CBC/PKCS5Padding", true, true, false, false),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, true, true),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, true, true),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, true, false),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, true, false),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, false, true),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, false, true),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, false, false),
+            Arguments.of("AES/CBC/PKCS5Padding", false, true, false, false)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("resetByteBufferTestParameters")
+    public void doTestResetByteBuffer(String algo, boolean encrypt, boolean direct, boolean updateFirst, boolean updateSecond) throws Exception {
+        // Instantiate algorithm and create appropriate key.
+        c = Cipher.getInstance(algo, getProviderName());
+        String a[] = algo.split("/");
+        KeyGenerator kg = KeyGenerator.getInstance(a[0], getProviderName());
+        key = kg.generateKey();
+        
+        // Create encrypted data.
+        c.init(Cipher.ENCRYPT_MODE, key, c.getParameters());
+        encrypted = new byte[c.getOutputSize(data.length)];
+        c.doFinal(data, 0, data.length, encrypted, 0);
+
+        // Initialize for encryption or decryption using byte buffers.
+        if (encrypt == true) {
+            initializeEncrypt(direct);
+        } else {
+            initializeDecrypt(direct);
+        }
+
+        // Perform first operation, either an update or dofinal.
+        if (updateFirst == true) {
+            doUpdate();
+        } else {
+            doFinal();
+        }
+
+        // Perform second operation, either an update or dofinal.
+        if (updateSecond == true) {
+            doUpdate();
+        } else {
+            doFinal();
+        }
```

Comment:
Should we also include an additional `doFinal()` in case 2 updates were performed?

Reply:
I don't believe that we want do finalize a double update. This test is just intended to make sure that no exceptions occur with various combinations of `doUpdate` and `doFinal` calls and that the output buffer sizes are calculated correctly.